recognition/include/pcl/recognition/impl/linemod/line_rgbd.hpp

```diff
@@ -150,72 +151,66 @@ LineRGBD<PointXYZT, PointRGBT>::loadTemplates (const std::string &file_name, con
 
   // Compute 3D bounding boxes from the template point clouds
   bounding_boxes_.resize (template_point_clouds_.size ());
-  for (std::size_t i = 0; i < template_point_clouds_.size (); ++i)
+  for (std::size_t i = 0; i < template_point_clouds_.size (); ++i) 
   {
-    PointCloud<PointXYZRGBA> & template_point_cloud = template_point_clouds_[i];
-    BoundingBoxXYZ & bb = bounding_boxes_[i];
-    bb.x = bb.y = bb.z = std::numeric_limits<float>::max ();
-    bb.width = bb.height = bb.depth = 0.0f;
+    bounding_boxes_[i] = computeBoundingBoxAndCenterTemplatePointCloud (template_point_clouds_[i]);
+  }
 
-    float center_x = 0.0f;
-    float center_y = 0.0f;
-    float center_z = 0.0f;
-    float min_x = std::numeric_limits<float>::max ();
-    float min_y = std::numeric_limits<float>::max ();
-    float min_z = std::numeric_limits<float>::max ();
-    float max_x = -std::numeric_limits<float>::max ();
-    float max_y = -std::numeric_limits<float>::max ();
-    float max_z = -std::numeric_limits<float>::max ();
-    std::size_t counter = 0;
-    for (std::size_t j = 0; j < template_point_cloud.size (); ++j)
-    {
-      const PointXYZRGBA & p = template_point_cloud.points[j];
+  return (true);
+}
 
-      if (!std::isfinite (p.x) || !std::isfinite (p.y) || !std::isfinite (p.z))
-        continue;
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointXYZT, typename PointRGBT> BoundingBoxXYZ
+pcl::LineRGBD<PointXYZT, PointRGBT>::computeBoundingBoxAndCenterTemplatePointCloud (PointCloud<PointXYZRGBA> & template_point_cloud)
+{
+  BoundingBoxXYZ bb;
+  bb.x = bb.y = bb.z = std::numeric_limits<float>::max ();
+  bb.width = bb.height = bb.depth = 0.0f;
+
+  Eigen::Vector3f geometric_center = Eigen::Vector3f::Zero ();
+  Eigen::Vector3f min_pos, max_pos;
+  min_pos.fill (std::numeric_limits<float>::max ());
+  max_pos.fill (std::numeric_limits<float>::lowest ());
+  std::size_t counter = 0;
+  for (std::size_t j = 0; j < template_point_cloud.size (); ++j) 
+  {
+    const PointXYZRGBA & p = template_point_cloud.points[j];
 
-      min_x = std::min (min_x, p.x);
-      min_y = std::min (min_y, p.y);
-      min_z = std::min (min_z, p.z);
-      max_x = std::max (max_x, p.x);
-      max_y = std::max (max_y, p.y);
-      max_z = std::max (max_z, p.z);
+    if (!isXYZFinite(p))
+      continue;
 
-      center_x += p.x;
-      center_y += p.y;
-      center_z += p.z;
+    min_pos.cwiseMin (p.getVector3fMap ());
+    max_pos.cwiseMax (p.getVector3fMap ());
```

Comment:
Shouldn't those two lines also assign? It seems that `cwiseMin` and `cwiseMax` don't modify any of the objects involved and return a result.

Reply:
My bad, I forgot these are not in-place method. Thanks for catching this!